### ISSUE TITLE
Cite using "shorthand"

### ISSUE THREAD

**Issue by ronaldtse** [riboseinc/asciidoctor-bibliography#15]
I have some `bib` files that provide the `shorthand` attribute. E.g.,

```latex
%ISO 31000:2009(E)
@techreport{iso:31000,
  author = {ISO},
  institution = {International Organization for Standardization},
  address = {Geneva, Switzerland},
  title = {Risk management---Principles and guidelines},
  number = {31000:2009},
  shorthand = {ISO 31000:2009},
  type = {ISO},
  year = {2009}
}

% ISO Guide 73:2009
@techreport{iso:guide73,
  author = {ISO},
  institution = {International Organization for Standardization},
  address = {Geneva, Switzerland},
  title = {Risk management---Vocabulary},
  number = {73:2009(E/F)},
  shorthand = {ISO Guide 73:2009},
  type = {ISO Guide},
  year = {2013}
}
```

I'd like to have a mode that provides a citation using the shorthand, e.g., "[ISO:31000:2009]", instead of "[16]". With this mode obviously not all bib infos have the "shorthand" which could fall back to "authoryear" or just pure numbers ("[16]").

**Comment by paolobrasolin** [comment]
Writing the ISO CSL style and using a switch to prefill the `citation-label` using `shorthand` is the best way to go.

**Comment by ronaldtse** [comment]
Agree. There does not seem to be an agreed practice on how to BibTeX cite ISO documents yet. Perhaps you can formalize a method based on ISO 690?

http://dominiodelasciencias.com/ojs/documentos/ISO690-2010.pdf

**Comment by ronaldtse** [comment]
@paolobrasolin is this issue still valid? Thanks

**Comment by paolobrasolin** [comment]
It is. I gave this some thought.

I think the best way to handle this kind of situation, since there is no fixed convention for BibTeX data structure, would be to introduce an option like
```
cite[iso:31000, by=shorthand]
```
that would produce an (hyperlinked if necessary) citation text `[ISO 21000:2009]`.
That is, we let the user specify an arbitrary (text) field to use as a citation body.
This has the bonus of solving some other weird edge cases too without introducing extra specialization.

Allowing more flexibility would probably be more fruitful than trying to enforce a convention, despite some extra verbosity (e.g. compare with a more rigid `citeiso[iso:31000]`).

Also, this is perfectly in line with the idea in #59.

My only doubts about the usability of this are about multiple citations and wrapping brackets.
What are your thought on this?

**Comment by ronaldtse** [comment]
@paolobrasolin I agree with this approach since the “by” keyword could also refer to other bibtex keys too. And in any case, the gem should support bibliography databases beyond just bibtex in the future!

**Comment by ribose-jeffreylau** [comment]
Hi, is this idea still being considered?  Asking cuz this would be something we would start using right away.
